python/cuml/linear_model/linear_regression.pyx

```diff
@@ -316,6 +317,16 @@ class LinearRegression(Base,
 
         return self
 
+    def predict_(self, X, convert_dtype=True) -> CumlArray:
+        self.dtype = self.coef_.dtype
+        self.n_cols = self.coef_.shape[0]
```

Comment:
these would be better as properties set in the fit method in general, no?

Reply:
These values are required for by the `LinearPredictMixin` independently of whether the estimator was trained on device or host. That's why they have to be recovered just in time for predictions from the `coef_` attribute.